python/tank/descriptor/io_descriptor/base.py

```diff
@@ -369,6 +379,31 @@ def _get_locally_cached_versions(self):
 
         return all_versions
 
+    def _check_minimum_python_version(self, manifest_data):
+        """
+        Checks if the current Python version meets the minimum required version
+        specified in the manifest data.
+
+        :param manifest_data: Dictionary containing bundle manifest/info.yml data
+        :returns: True if current Python version is compatible, False otherwise
+        """
+        # Get current Python version as string (e.g., "3.9.13")
+        current_version_str = ".".join(str(i) for i in sys.version_info[:3])
```

Comment:
I think this can be cleaner. Not sure if this is supported by 3.9
```suggestion
        major, minor, micro, *_ = sys.version_info
        current_version_str =  f"{major}.{minor}.{micro}"
```